sheepdog/api.py

```diff
@@ -162,6 +163,12 @@ def app_init(app):
 
 
 # Setup logger
+app.logger.setLevel(
+    logging.DEBUG if (os.environ.get("GEN3_DEBUG") == "True") else logging.WARNING
+)
+app.logger.propagate = False
+while app.logger.handlers:
+    app.logger.removeHandler(app.logger.handlers[0])
```

Comment:
`app.logger.propagate = False` was necessary here to prevent log messages from being duplicated.

I don't think `removeHandler` is absolutely necessary - when debugging this locally, `app.loggers.handlers` is empty just before `while app.logger.handlers:`:
```
(sheepdog) bash-3.2$ GEN3_DEBUG=True python run.py
/Users/johnmccann/.local/share/virtualenvs/sheepdog-mo8Kq6VZ/lib/python3.7/site-packages/authlib/client/__init__.py:12: AuthlibDeprecationWarning: Deprecate "authlib.client", USE "authlib.integrations.requests_client" instead.
It will be compatible before version 1.0.
Read more <https://git.io/Jeclj#file-rn-md>
  deprecate('Deprecate "authlib.client", USE "authlib.integrations.requests_client" instead.', '1.0', 'Jeclj', 'rn')
> /Users/johnmccann/Programs/sheepdog/sheepdog/api.py(171)<module>()
-> while app.logger.handlers:
(Pdb) app.logger.handlers
[]
(Pdb)
```
However, in the [Flask documentation](https://flask.palletsprojects.com/en/1.1.x/logging/), there is mention of a default handler that gets configured:

"If `app.logger` is accessed before logging is configured, it will add a default handler."

I didn't think it was entirely clear what would constitute the logger being "accessed" (e.g in the lines preceding `while app.logger.handlers:` above, although I would have thought that `app.logger.setLevel(...)` and `app.logger.propagate = False` would mean that the logger has been "accessed", there is still no default handler configured on line 170)

To avoid a situation in which we potentially add a second handler when Flask already has a default one configured and log messages get duplicated as a result, I thought it was safer to just explicitly clear out all handlers.